types.go

```diff
@@ -49,9 +49,10 @@ func newNumericDateFromSeconds(f float64) *NumericDate {
 // MarshalJSON is an implementation of the json.RawMessage interface and serializes the UNIX epoch
 // represented in NumericDate to a byte array, using the precision specified in TimePrecision.
 func (date NumericDate) MarshalJSON() (b []byte, err error) {
+	prec := int(math.Log10(float64(time.Second) / float64(TimePrecision)))
```

Comment:
A minor thing here: We probably need to also set a minimum precision to 0 because if you set it to TimePrecision of Seconds, log10 will return -1 which means unlimited precision. Otherwise we could return an error if you set precision above seconds (which does not really make sense anyway).

Reply:
Good catch. I hope no one is setting the precision to minutes or higher. This is something the specification could have been more explicit about.